Model/FrameModel.py

```diff
@@ -86,38 +81,36 @@ def _update_view_screen_mobs(self, ftime, frame):
             self._data_queue_received.append(frame_pkg)
         self._current_frame = frame
         self._update_view_screen_ball()
-        #if frame.detection.camera_id == 1:
-        #    print(frame.detection.robots_yellow)
+
         self._update_view_screen_robot('blue')
         self._update_view_screen_robot('yellow')
 
     def _update_view_screen_ball(self):
         """ Mise à jour des données de la vue de la balle """
-        try:
-            x = self._current_frame.detection.balls[0].x
-            y = self._current_frame.detection.balls[0].y
-            self._controller.set_ball_pos_on_screen(x, y)
-        except Exception as e:
-            self._controller.hide_mob()
+        balls = self._current_frame.detection.balls
+        if len(balls) > 0:
+            self._last_frame_with_ball_time = datetime.now()
+            self._controller.set_ball_pos_on_screen(balls[0].x, balls[0].y)
+
+        # Only one frame out of 4 have the ball's geometry
```

Comment:
Not always true. The ball can be in multiple camera.

Reply:
The point of the comment is that in the old version, you would need to have the ball in all 4 cameras to prevent flickering.